### PR TITLE
added shoppingcart + ticket DAO and service

### PR DESCRIPTION
**Pull request by Vosker**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            Query<ShoppingCart> getCartByUserQuery
+                    = session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "where sc.user = :user", ShoppingCart.class);
+            getCartByUserQuery.setParameter("user", user);
+            return getCartByUserQuery.getSingleResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get shopping cart of user "
+                    + user, e);
+        }
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/impl/ShoppingCartDaoImpl.java:50]: No need to catch this

File: src/main/java/com/dev/cinema/dao/impl/TicketDaoImpls.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(ticket);
+            transaction.commit();
+            return ticket;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add the shopping cart "
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dao/impl/TicketDaoImpls.java]: Ctrl + C, Ctrl + V, huh? 😄

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartDao.add(shoppingCart);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        shoppingCart.getTickets().clear();
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: Won't work in the next task
Let's just pass an empty list using the setTickets() method

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

**Comment by dariiabutorina** [comment]
Travis died, let's not be criminals

File: src/main/java/com/dev/cinema/dao/impl/TicketDaoImpls.java (hunk 2 above)
Comment (Vosker) [reply src/main/java/com/dev/cinema/dao/impl/TicketDaoImpls.java]: +)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**